run_bin_test.go

```diff
@@ -396,6 +398,55 @@ func TestCoverageCollector_writeArgs(t *testing.T) {
 	}
 }
 
+func TestPreAndPostExec(t *testing.T) {
+	tests := []struct {
+		name         string
+		preCmdFuncs   int
+		postCmdFuncs  int
+	}{
+		{
+			name:	"add two preCmdFuncs and one postCmdFunc",
```

Comment:
I don't know if it makes sense to test like this. The pre and post command funcs aren't really coupled in behavior, so I don't think testing combinations is necessary.

Reply:
I would also test not just the presence, but the behavior. Maybe add a function that outputs to a file and test that it works (maybe there's an even simpler test case).

Reply:
yeah that totally makes sense - updated tests to have pre/post cmd funcs output to a buffer and tests verify the output.